responses/tests/test_recorder.py

```diff
@@ -15,15 +17,126 @@
     # python 3.11
     import tomllib as _toml  # type: ignore[no-redef]
 
+_NOT_CARE = object()
+
+
+class _CompareDict:
+    def __init__(self, obj):
+        self.obj = obj
+
+    def __eq__(self, other):
+        if self.obj is _NOT_CARE:
+            return True
+        elif isinstance(self.obj, collections.abc.Mapping):
+            if not isinstance(other, collections.abc.Mapping):
+                return False
+            if sorted(self.obj.keys()) != sorted(other.keys()):
+                return False
+            for key in self.obj.keys():
+                if _CompareDict(self.obj[key]) != other[key]:
+                    return False
+            return True
+        elif isinstance(self.obj, list):
+            if not isinstance(other, list):
+                return False
+            if len(self.obj) != len(other):
+                return False
+            for i, (obj_item, other_item) in enumerate(zip(self.obj, other)):
+                if _CompareDict(obj_item) != other_item:
+                    return False
+            return True
+        else:
+            return self.obj == other
```

Comment:
why do we need all of those?

Reply:
Because the unittest need to compare the full dict data.

In the new version, headers are added into the recorded data, which may contain something like `Python/3.x.y` or something else which are not important for unittesting. So the `_CompareDict` and `_NOT_CARE` object are added to ignore these values.